### ISSUE TITLE
Implementing convexity using Hessian Matrix

### ISSUE THREAD

**Issue by Emilius12** [sympy/sympy#26427]
I saw in #15776 that `is_convex` cannot be used on multivariate functions.

I think it is possible to implement it using `hessian` and `is_positive_semidefinite` (see [this Wikipedia article](https://en.wikipedia.org/wiki/Convex_function)). 

In fact, I gave it a few try and it worked fine, but I think some more testing is required. It is also mandatory to keep the disjunction between univariate and multivariate because `hessian` does not support univariate functions. I'm willing to contribute to the implementation and testing of this feature. However, I'd appreciate some feedback before proceeding further.

**Version** : sympy 1.12

**Comment by sylee957** [comment]
What is the problem with univariate function with hessian? For the univariate function, the hessian would be a matrix with single element, and if it is positive definite, it would trivially logically reduce to its second derivative being positive.

**Comment by Emilius12** [comment]
> What is the problem with univariate function with hessian? For the univariate function, the hessian would be a matrix with single element, and if it is positive definite, it would trivially logically reduce to its second derivative being positive.


I agree that it would be nice to have a function that works for both cases but the issue is the `hessian` function from sympy, which does not support univariate function. I tried to modify the definition of this function but it made many tests fail.
